### PR TITLE
Add unit tests for camera

### PR DESCRIPTION
**Pull request by dijidiji**
# Description

This adds tests for the camera functions in camera.cpp.  Duplicate of #119 but without the tag changes.

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)

## Testing Checklist

- [x] Tested with skunit_tests

I've verified that the tests run and assertions pass as expected. I've also run code coverage which reports 80% coverage for camera.cpp

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from ... on the Pull Request

### CODE REVIEW

**Review by Broccoli811 — APPROVED** [review]
The new test file for the camera module looks solid. It covers a broad range of key functionalities for a camera function, and it is well documented and readable to me.

All tests passed successfully in my environment, and the expected results match the documented behavior in camera.cpp

A suggestion that could be useful is to add edge case tests for when the camera is moved to negative/unreasonable coordinates or when the rectangles lie exactly on the screen border. This may ensure behavior is consistent in those scenarios.

Also, you may want to update the company board card on your additions/tests and potentially reference the camera.cpp document, it may sound trivial to some, but it doesn't hurt to add more context to the card for the next reviewer :)

Overall, great implementation!

File: coresdk/src/test/unit_tests/unit_test_camera.cpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+class CameraTestFixture {
+    public:
+        CameraTestFixture()
+        {
+            set_camera_position(point_at(0.0, 0.0));
+        }
+};
+
+
+TEST_CASE_METHOD(CameraTestFixture, "camera position correct after moving", "[set_camera_position]")
```
Comment (Broccoli811) [inline coresdk/src/test/unit_tests/unit_test_camera.cpp:23]: Maybe consider also testing with negative or very large coordinates to cover edge cases

File: coresdk/src/test/unit_tests/unit_test_camera.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+    SECTION("screen point to world space")
+    {
+        point_2d result = to_world(point_at(100.0, 100.0));
+        REQUIRE_THAT(result.x, WithinRel(250.0));
+        REQUIRE_THAT(result.y, WithinRel(250.0));
+    }
+}
+
+TEST_CASE_METHOD(CameraTestFixture, "check if rectangle is on screen", "[rect_on_screen][rect_in_window]")
```
Comment (Broccoli811) [inline coresdk/src/test/unit_tests/unit_test_camera.cpp]: Perhaps consider testing on the border of the screen or partially in the screen to check the boundary behavior of the rectangle

File: coresdk/src/test/unit_tests/unit_test_camera.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        }
+        SECTION("rectangle is out of given window")
+        {
+            REQUIRE_FALSE(rect_in_window(wind, rect));
+        }
+    }
+    close_window(wind);
+}
+
+TEST_CASE_METHOD(CameraTestFixture, "check if point is on screen", "[point_on_screen][point_in_window]")
```
Comment (Broccoli811) [inline coresdk/src/test/unit_tests/unit_test_camera.cpp]: Same deal here with boundary tests. What if the point was on the edge of the screen?

**Review by JPF2209 — APPROVED** [review]
## Type of Change
- This piece of code correctly identifies the changes made being adding additional tests

## Code Readability
- The code is very understandable and clear in what it needs to be comparing it to the other code.

## Maintainability
- As this is in the same style as the other code, it's quite maintainable being simple while doing the job.

## Code Simplicity
- This code is simple in it's execution following established design patterns and best practices 

## Edge Cases
- The code deals with edge cases simply since the inner functionality of the test code manages this

## Test Thoroughness
- For all the key tests in this type of scenario, this covers it all and the code has been tested in multiple environments and it works.

## Backward Compatibility
- The code doesn't break existing functionality in the way it works.

## Performance Considerations
- Performance is usually more of a concern with the function itself but this works out well for performance.

## Security Concerns
- Ensure to fulfilll branch protection policy

## Dependencies
- There are no new added dependencies. 

## Documentation
- The documentation provided is through and simple to follow at the same time

File: coresdk/src/test/unit_tests/unit_test_camera.cpp (hunk 1 above)
Comment (dijidiji) [reply coresdk/src/test/unit_tests/unit_test_camera.cpp:23]: Added a negative coordinates test. Not sure about how best to test large coordinates because it could be arbitrarily large up to the maximum value of a double

File: coresdk/src/test/unit_tests/unit_test_camera.cpp (hunk 2 above)
Comment (dijidiji) [reply coresdk/src/test/unit_tests/unit_test_camera.cpp]: Added tests for cases where the rectangle touches the border of the screen and where it's partially on screen. Also split into two test cases for readability

File: coresdk/src/test/unit_tests/unit_test_camera.cpp (hunk 3 above)
Comment (dijidiji) [reply coresdk/src/test/unit_tests/unit_test_camera.cpp]: Added tests for when a point is on the edge of the screen and separated on screen/in window test cases

**Review by monicavtasmin — APPROVED** [review]
